Sources/PayPalNativeCheckout/PayPalClient.swift

```diff
@@ -13,60 +9,67 @@ public class PayPalClient {
+    init(config: CoreConfig, checkoutFlow: CheckoutProtocol) {
         self.config = config
-        self.CheckoutFlow = checkoutFlow
+        self.nativeCheckout = checkoutFlow
     }
 
     /// Present PayPal Paysheet and start a PayPal transaction
     /// - Parameters:
     ///   - request: the PayPalRequest for the transaction
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
     ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
-    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) {
-        CheckoutFlow.set(config: config)
-
-        CheckoutFlow.start(
-            presentingViewController: presentingViewController,
-            createOrder: { order in
-                order.set(orderId: request.orderID)
-            },
-            onApprove: { approval in
-                self.notifySuccess(for: approval)
-            },
-            onCancel: {
-                self.notifyCancellation()
-            },
-            onError: { errorInfo in
-                self.notifyFailure(with: errorInfo)
-            }
-        )
+    public func start(presentingViewController: UIViewController?, orderID: String, deleagate: PayPalDelegate?){
+        self.delegate = deleagate
+        DispatchQueue.main.async {
+            self.nativeCheckout.start(
+                presentingViewController: presentingViewController,
+                createOrder: { order in
+                    order.set(orderId: orderID)
+                },
+                onApprove: { approval in
+                    self.notifySuccess(for: approval)
+                },
+                onShippingChange: { shippingChange, shippingChangeAction in
+                    self.notifyShippingChange(shippingChange: shippingChange, shippingChangeAction: shippingChangeAction)
+                },
+                onCancel: {
+                    self.notifyCancellation()
+                },
+                onError: {error in
+                    self.notifyFailure(with: error)
+                }
+            )
+        }
     }
 
-    private func notifySuccess(for approval: PayPalCheckoutApprovalData) {
-        let payPalResult = PayPalResult(orderID: approval.ecToken, payerID: approval.payerID)
-        delegate?.paypal(self, didFinishWithResult: payPalResult)
+    private func notifySuccess(for approval: PayPalCheckout.Approval) {
+        delegate?.paypal(didFinishWithResult: approval)
     }
 
     private func notifyFailure(with errorInfo: PayPalCheckoutErrorInfo) {
         let error = PayPalError.nativeCheckoutSDKError(errorInfo)
-        delegate?.paypal(self, didFinishWithError: error)
+        delegate?.paypal(didFinishWithError: error)
     }
 
     private func notifyCancellation() {
-        delegate?.paypalDidCancel(self)
+        delegate?.paypalDidCancel()
+    }
```

Comment:
Same [question](https://github.com/paypal/iOS-SDK/pull/69/files#r941815212) for this helper function